### ISSUE TITLE
p-password: input not connected with label (a11y)

### ISSUE THREAD

**Issue by lauraLuisa** [primefaces/primeng#13952]
### Describe the bug

Label tag, aria-labelledby or aria-label are not correctly associated with input element of p-password component as described in the [Accessibility section](https://primeng.org/password#accessibility) of the p-password documentation.

### Environment

Angular 16, PrimeNg 16 

### Reproducer

https://stackblitz.com/edit/github-bxn4ym?file=src%2Fapp%2Fapp.component.html

### Angular version

16.2.6

### PrimeNG version

16.3.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.1

### Steps to reproduce the behavior

1. link a `p-password` component to a label (via label tag, aria-labelledby or aria-label)
2. Select the input element via the browser console
3. Switch to accessibility tab an see that the input element has no accessible name
![Bildschirmfoto 2023-10-25 um 09 33 07](https://github.com/primefaces/primeng/assets/67911741/e5a6a7a3-d7fe-46fc-b6c0-594617bdd836)
or run a tool like google lighthouse (accessibility) an see that there is an issue because the input field has no associated label.

### Expected behavior

The input element which lies under the p-password component is correctly linked to the label.

**Comment by SoyDiego** [comment]
Hi @lauraLuisa, thanks for report the problem.
The problem is in the Documentation. I have created a PR fixing the problem.
You can check here #13973.

The correct way is using `ariaLabel` and `ariaLabelledBy` instead of `aria-label`and `aria-labelled-by`.
Example:

```
  <div>
    <span id="pwd2">Password</span>
    <p-password ariaLabelledBy="pwd2"></p-password>
  </div>

  <div>
    <p-password ariaLabel="Password"></p-password>
  </div>
```

![image](https://github.com/primefaces/primeng/assets/19764334/7a3b41c8-3d2f-44dc-8b80-6127623986dd)
![image](https://github.com/primefaces/primeng/assets/19764334/b214bf03-6261-42fa-b076-1096bf932682)

Here you have the correct properties:
![image](https://github.com/primefaces/primeng/assets/19764334/fcb0dbb3-245f-4b52-8d6d-30a6b23d0ee0)

But yes, the problem was in the documentation.

Thanks!

**Comment by heeae** [comment]
I have another issue related to password component, which I cannot specify the aria-describedBy (tag not available) 
and aria-required (it has this variable,  but not applied to rendered input field) 

which I cannot specify the description and required flag to password related field

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
